Objects/dictobject.c

```diff
@@ -3919,24 +3919,92 @@ dictviews_sub(PyObject* self, PyObject *other)
     return result;
 }
 
+static int
+dictitems_contains(_PyDictViewObject *dv, PyObject *obj);
+
 PyObject*
 _PyDictView_Intersect(PyObject* self, PyObject *other)
 {
-    PyObject *result = PySet_New(self);
-    PyObject *tmp;
-    _Py_IDENTIFIER(intersection_update);
+    PyObject *result;
+    PyObject *it;
+    PyObject *key;
+    Py_ssize_t len_self;
+    int rv;
+    int (*dict_contains)(_PyDictViewObject *, PyObject *);
```

Comment:
Please define local variables when they are initialized.

For example

```
int rv = dict_contains((_PyDictViewObject *)self, key);
```

Reply:
It is also okay to list them all at the top.